src/components/activity-groups-list/activity-groups-list.tsx

```diff
@@ -1,17 +1,28 @@
-import React, { FC, useMemo } from 'react';
-import { SectionList, Text, View } from 'react-native';
-
-import { emptyFn } from '../../config/general';
-import { useFakeRefreshControlProps } from '../../hooks/use-fake-refresh-control-props.hook';
-import { ActivityGroup, emptyActivity } from '../../interfaces/activity.interface';
-import { isTheSameDay, isToday, isYesterday } from '../../utils/date.utils';
-import { DataPlaceholder } from '../data-placeholder/data-placeholder';
-import { OptimalPromotionItem } from '../optimal-promotion-item/optimal-promotion-item';
-import { RefreshControl } from '../refresh-control/refresh-control';
+import { FlashList, ListRenderItem } from '@shopify/flash-list';
+import React, { FC, useCallback, useMemo } from 'react';
+import { Text, View } from 'react-native';
+
+import { DataPlaceholder } from 'src/components/data-placeholder/data-placeholder';
+import { OptimalPromotionItem } from 'src/components/optimal-promotion-item/optimal-promotion-item';
+import { RefreshControl } from 'src/components/refresh-control/refresh-control';
+import { emptyFn } from 'src/config/general';
+import { useFakeRefreshControlProps } from 'src/hooks/use-fake-refresh-control-props.hook';
+import { ActivityGroup, emptyActivity } from 'src/interfaces/activity.interface';
+import { isTheSameDay, isToday, isYesterday } from 'src/utils/date.utils';
+
 import { ActivityGroupItem } from './activity-group-item/activity-group-item';
 import { ActivityGroupsListSelectors } from './activity-groups-list.selectors';
 import { useActivityGroupsListStyles } from './activity-groups-list.styles';
 
+type ListItem = string | ActivityGroup;
+
+const keyExtractor = (item: ListItem) => (typeof item === 'string' ? item : item[0].hash);
+const getItemType = (item: ListItem) => (typeof item === 'string' ? 'sectionHeader' : 'row');
+
+const ListEmptyComponent = <DataPlaceholder text="No Activity records were found" />;
+
+const AVERAGE_ITEM_HEIGHT = 150;
```

Comment:
Did you mean `Math.floor(formatSize(150))`?

Reply:
docs for `estimatedItemSize` prop: 
If the items have a lot of different sizes, choose a few that are pretty different, use element inspector, read their sizes, and calculate an average. This average will work great, don't worry about solving for different devices sizes. There's enough tolerance internally.

so i think there is no need of `formatSize` and `Math.floor` in this case

Reply:
also flashList itself gives us this value in the console, if `estimatedItemSize` prop in not set